lntest/itest/lnd_test.go

```diff
@@ -8751,6 +8751,24 @@ func testNodeAnnouncement(net *lntest.NetworkHarness, t *harnessTest) {
 		t.Fatalf("unable to connect bob to carol: %v", err)
 	}
 
+	// We'll connect Alice with Dave now to ensure she receives the node
```

Comment:
Feels a bit like working around the actual problem. Does this mean a node connecting to another node never will receive its node announcement if no new channel is opened? Shouldn't this be sent during the historical sync?

Reply:
> Does this mean a node connecting to another node never will receive its node announcement if no new channel is opened? 

That is part of the spec, see https://github.com/lightningnetwork/lightning-rfc/blob/master/07-routing-gossip.md#the-node_announcement-message.

> Shouldn't this be sent during the historical sync?

Alice has been connected to other nodes prior to this test, so a historical sync does not cover it.

Reply:
> Alice has been connected to other nodes prior to this test, so a historical sync does not happen.

Should Alice receive the announcement from the node she is already connected to (Bob?) then. If she has no connections before connecting to Dave, she should perform a historical sync.

Reply:
You're right, Alice ends up not receiving Dave's announcement from Bob due to https://github.com/lightningnetwork/lnd/issues/3058.

Reply:
I think we still want a connection to be made explicitly between Alice and Bob at this point however to avoid relying on the test harness connecting them before the start of every test.